src/Note.cpp

```diff
@@ -0,0 +1,378 @@
+        return true;
+    }
+    if (is_name("all", to_list_)) {
+        return true;
+    }
+    if (ch.is_immortal() && is_name("immortal", to_list_)) {
+        return true;
+    }
+    if (is_name(ch.name, to_list_)) {
+        return true;
+    }
+    return false;
+}
+
+bool Note::sent_by(const Char &ch) const { return matches(ch.name, sender_); }
+
+void Note::save(FILE *file) const {
+    fmt::print(file, "Sender {}~\n", sender_);
+    fmt::print(file, "Date {}~\n", date_);
+    fmt::print(file, "Stamp {}\n", Clock::to_time_t(date_stamp_));
+    fmt::print(file, "To {}~\n", to_list_);
+    fmt::print(file, "Subject {}~\n", subject_);
+    fmt::print(file, "Text\n{}~\n", text_);
+}
+
+void Notes::erase(const Note &note) {
+    auto it = ranges::find_if(notes_, [&](const Note &data) { return &data == &note; });
+    if (it != notes_.end())
+        notes_.erase(it);
+}
+Note &Notes::add(Note note) { return notes_.emplace_back(std::move(note)); }
+
+int Notes::num_unread(const Char &ch) const {
+    int notes = 0;
+    for (auto &note : notes_) {
+        if (note.is_to(ch) && !note.sent_by(ch) && note.date_stamp() > ch.last_note) {
+            notes++;
+        }
+    }
+    return notes;
+}
+
+Note *Notes::lookup(int index, const Char &ch) {
+    for (auto &note : notes_) {
+        if (note.is_to(ch)) {
+            if (index-- <= 0) {
+                return &note;
+            }
+        }
+    }
+    return nullptr;
+}
+
+Note *Notes::lookup(Time date, const Char &ch, int *index) {
+    int count = 0;
+
+    for (auto &note : notes_) {
+        if (note.is_to(ch) && !note.sent_by(ch)) {
+            if (date < note.date_stamp()) {
+                *index = count;
+                return &note;
+            }
+            count++;
+        }
+    }
+    return nullptr;
+}
+
+void Notes::save(FILE *file) const {
+    for (auto &note : notes_)
+        note.save(file);
+}
+
+static Note &ensure_note(Char &ch) {
+    if (!ch.pnote)
+        ch.pnote = std::make_unique<Note>(ch.name);
+    return *ch.pnote;
+}
+
+Note Note::from_file(FILE *fp) {
+    auto expect = [&](std::string_view expected) {
+        auto *word = fread_word(fp);
+        if (!matches(word, expected))
+            throw std::runtime_error(fmt::format("Expected '{}'", expected));
+    };
+    auto expect_str = [&](std::string_view expected) {
+        expect(expected);
+        return fread_stdstring(fp);
+    };
+    Note note(expect_str("sender"));
+    note.date_ = expect_str("date");
+    expect("stamp");
+    note.date_stamp_ = Clock::from_time_t(fread_number(fp));
+    note.to_list_ = expect_str("to");
+    note.subject_ = expect_str("subject");
+    note.text_ = expect_str("text");
+    return note;
+}
+
+void Note::subject(std::string_view subject) { subject_ = subject; }
+void Note::date(Time point) {
+    date_ = formatted_time(point);
+    date_stamp_ = point;
+}
+void Note::to_list(std::string_view to_list) { to_list_ = to_list; }
```

Comment:
I'd been wondering if there ought to be some sort of limit on how many recipients you can specify. We have the regular max input length, I guess that covers it.  A bonus thought,  the to list (from ArgParser) could be uniq'd too :)

Reply:
Right :)